src/components/PeopleTable/PeopleTable.tsx

```diff
@@ -0,0 +1,89 @@
+import { useEffect, useState, useMemo } from 'react';
+import { Person } from '../../types';
+import { Loader } from '../Loader';
+import { getPeople } from '../../api';
+import { PersonItem } from '../PersonItem';
+import React from 'react';
+
+export const PeopleTable = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+  const isPeopleEmpty = people.length === 0;
+
+  const preparePeopleWithLinks = (peopleList: Person[]) => {
+    return peopleList.map(person => {
+      const { fatherName, motherName } = person;
+
+      return {
+        ...person,
+        fatherNameLink:
+          peopleList.find(p => p.name === fatherName)?.slug || null,
+        motherNameLink:
+          peopleList.find(p => p.name === motherName)?.slug || null,
+      };
+    });
+  };
+
+  const preparedPeople = useMemo(
+    () => preparePeopleWithLinks(people),
+    [people],
+  );
+
+  useEffect(() => {
+    setIsLoading(true);
+    getPeople()
+      .then(setPeople)
+      .then(() => {
+        setIsLoading(false);
+      })
+      .catch(() => {
+        setIsError(true);
+        setIsLoading(false);
+      });
+  }, []);
+
+  return (
+    <div className="container">
+      <div className="block">
+        <div className="box table-container">
+          {isLoading && <Loader />}
+
+          {isError && (
+            <p data-cy="peopleLoadingError" className="has-text-danger">
+              Something went wrong
+            </p>
+          )}
+
+          {!isLoading && isPeopleEmpty && (
```

Comment:
<img width="951" alt="image" src="https://github.com/mate-academy/react_people-table-basics/assets/86789161/b016e1bf-6733-4e65-a94f-90ba93c75da6">

2 messages are displayed at the same time

Reply:
This comment is not fixed